test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatConfigTests.cs

```diff
@@ -218,6 +219,117 @@ public void ConfigGetConfigKeyCommand_HelpMessage_Success()
             DotnetCliUtil.VerifyResultSuccess(result, helpMessage);
         }
 
+        [Theory]
+        [InlineData("signatureValidationMode", "accept")]
+        [InlineData("maxHttpRequestsPerSource", "64")]
+        public void ConfigSetCommand_WithConfigFileArg_AddsSetting(string key, string value)
+        {
+            // Arrange & Act
+            using var testInfo = new TestInfo("NuGet.Config");
+            var filePath = Path.Combine(testInfo.WorkingPath, "NuGet.Config");
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
+                waitForExit: true);
+
+            ISettings settings = Configuration.Settings.LoadDefaultSettings(
+                testInfo.WorkingPath,
+                configFileName: null,
+                machineWideSettings: new XPlatMachineWideSetting());
+
+            var configSection = settings.GetSection("config");
+            var values = configSection.Items.Select(c => c as AddItem).Where(c => c != null).ToList();
+            var configItems = values.Where(i => i.Key == key);
+            var configFilePath = configItems.Single().ConfigPath;
+
+            // Assert
+            Assert.Equal(0, result.ExitCode);
+            Assert.Equal(value, configItems.Single().Value);
+            Assert.Equal(filePath, configFilePath);
+        }
+
+        [Fact]
+        public void ConfigSetCommand_WithConfigFileArg_UpdatesSetting()
+        {
+            // Arrange & Act
+            using var testInfo = new TestInfo("NuGet.Config");
+            var key = "http_proxy";
+            var value = "http://company-octopus:8765@contoso.test";
+            var filePath = Path.Combine(testInfo.WorkingPath, "NuGet.Config");
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
```

Comment:
I'm not following how this test is different than `ConfigSetCommand_WithConfigFileArg_AddsSetting`.
Was the intention to have an existing `key` entry, then update that entry with `value`? I'm not sure this test is doing that, currently. If it is, please explain.

Reply:
Yes, the `TestInfo` used by this test has an existing `http_proxy` key, and the value is then updated during the test. Would it be better to get the current value of `http_proxy`, change it and get the new value, and then compare them?